doc/melange.rst

```diff
@@ -4,49 +4,188 @@
+  - ``commonjs`` will follow `CommonJS modules <https://nodejs.org/api/modules.html>`_,
+    and will produce imports like ``require('foo')`` and exports like ``module.exports = ....``
+
+``<optional-fields>`` are:
+
+- ``(entries <modules>)`` specifies what modules will be built with Melange. By
+  default, Dune will use all the ``.ml/.re`` files in the same directory as the
+  ``dune`` file. This includes ones present in the file system as well
+  as ones generated by user rules. You can restrict this list by using a
+  ``(entries <modules>)`` field. ``<modules>`` uses the :ref:`ordered-set-language`,
+  where elements are module names and don't need to start with an uppercase
+  letter. For instance, to exclude module ``Foo``, use ``(entries (:standard \
+  foo))``
+
+- ``(libraries <library-dependencies>)`` specifies the library's dependencies.
+  Melange libraries can only use the simple form, like
+  ``(libraries foo pkg.bar)``. Keep in mind the following limitations:
+  - The ``re_export`` and ``select`` forms are not
+  supported.
+
+- ``(package <package>)`` allows to define to which JavaScript package the
+  artifacts produced by the ``melange.emit`` stanza will belong to. This is
+  useful when the ``libraries`` field includes a public library, as it helps
+  making  sure the modules from the library and the entry modules that are part
+  of the ``melange.emit`` stanza belong to the same resulting JavaScript
+  package.
+
+- ``(preprocess <preprocess-spec>)`` specifies how to preprocess files when
+  needed. The default is ``no_preprocessing``, and other options are described
+  in the :ref:`preprocessing-spec` section.
+
+- ``(preprocessor_deps (<deps-conf list>))`` specifies extra preprocessor
+  dependencies preprocessor, i.e., if the preprocessor reads a generated file.
+  The specification of dependencies is described in the :ref:`deps-field`
+  section.
+
+- ``(flags <flags>)`` specifies compilation flags to be passed to ``melc``, the
+  command line tool for the Melange compiler that produces the ``.js`` artifacts.
+  ``<flags>`` is specified in the :ref:`ordered-set-language`. It also supports
+  ``(:include ...)`` forms. The default value for is taken from the environment,
+  as a result it's recommended to write ``(flags ...)`` fields as
+  follows: ``(flags (:standard <my options>))``
+
+Limitations
+===========
+
+- JavaScript "monorepos" are not supported, so the use case of having multiple
+  JavaScript packages produced by Melange in a single project (i.e.
+  `Yarn <https://classic.yarnpkg.com/blog/2017/08/02/introducing-workspaces/>`_
+  or `npm <https://docs.npmjs.com/cli/v7/using-npm/workspaces>`_ workspaces) is
+  not guaranteed to work.
+
+- Libraries that are placed above the directory where the ``melange.emit``
+  stanza is located can not be used, due to the file layout requirements
+  mentioned above in the ``target`` field description.
+
+- Libraries that are part of the installed world are not supported as
```

Comment:
what's the installed world? "Findlib libraries"?

Reply:
Yes. I wanted to express that we don't yet support installing Melange libraries with opam. Maybe the docs should just say that.